subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -101,11 +105,21 @@ public DefaultSettings(
         this.settingsScript = settingsScript;
         this.startParameter = startParameter;
         this.services = serviceRegistryFactory.createFor(this);
-        this.rootProjectDescriptor = createProjectDescriptor(null, settingsDir.getName(), settingsDir);
+        this.rootProjectDescriptor = createProjectDescriptor(null, getProjectName(settingsDir), settingsDir);
         this.dependencyResolutionManagement = services.get(DependencyResolutionManagementInternal.class);
         this.toolchainManagement = services.get(ToolchainManagementInternal.class);
     }
 
+    private static String getProjectName(File settingsDir) {
+        if(contains(File.listRoots(), settingsDir)) {
+            String rootIndicator = settingsDir.toPath().getRoot().toString().replaceAll("[\\\\:\\/]*", "");
+            // using "-" to separate the parts of the root project name to allow easier usage in the CLI, just in case.
+            return "generated-" + rootIndicator  + (rootIndicator.isEmpty() ? "" : "-") +
+                sha512().hashString(now().toString()).toString().substring(0, 6);
```

Comment:
The disadvantage here is that this name is unstable and will vary between builds. Did you think of a way to make it stable?

Reply:
not too much.
@ljacomet even suggested feeling some pain if you not follow the guidelines helps ;)
and the guideline is: name your root project.
also it is a rare corner case to my understanding. It will only be used in situations where the project is actually checked out on `root/c:\/d:\/...`